src/mender-update/context/context.cpp

```diff
@@ -387,66 +387,65 @@ expected::ExpectedBool MenderContext::MatchesArtifactDepends(const artifact::Hea
 		return expected::unexpected(ex_provides.error());
 	}
 	auto &provides = ex_provides.value();
-	return ArtifactMatchesContext(
-		provides, ex_dev_type.value(), hdr_view.header_info, hdr_view.type_info);
+	return ArtifactMatchesContext(provides, ex_dev_type.value(), hdr_view);
 }
 
 expected::ExpectedBool ArtifactMatchesContext(
-	const ProvidesData &provides,
-	const string &device_type,
-	const artifact::HeaderInfo &hdr_info,
-	const artifact::TypeInfo &type_info) {
-	if (!common::MapContainsStringKey(provides, "artifact_name")) {
+	const ProvidesData &provides, const string &device_type, const artifact::HeaderView &hdr_view) {
+	using common::MapContainsStringKey;
+	if (!MapContainsStringKey(provides, "artifact_name")) {
 		return expected::unexpected(
 			MakeError(ValueError, "Missing artifact_name value in provides"));
 	}
 
-	const auto &hdr_depends = hdr_info.depends;
-	AssertOrReturnUnexpected(hdr_depends.device_type.size() > 0);
-	if (!common::VectorContainsString(hdr_depends.device_type, device_type)) {
+	auto hdr_depends = hdr_view.GetDepends();
+	AssertOrReturnUnexpected(
+		MapContainsStringKey(hdr_depends, "device_type")
+		&& (hdr_depends["device_type"].size() > 0));
+	if (!common::VectorContainsString(hdr_depends["device_type"], device_type)) {
 		log::Error("Artifact device type doesn't match");
 		return false;
 	}
+	hdr_depends.erase("device_type");
 
-	if (hdr_depends.artifact_name) {
-		AssertOrReturnUnexpected(hdr_depends.artifact_name->size() > 0);
+	if (MapContainsStringKey(hdr_depends, "artifact_name")) {
+		AssertOrReturnUnexpected(hdr_depends["artifact_name"].size() > 0);
```

Comment:
I'm probably missing something, but now that you are matching `provides` against `hdr_depends` everywhere, why do we need these special cases at all?